src/js/Parser.js

```diff
@@ -0,0 +1,53 @@
+import { Station } from './Station';
+import { SubwayLine } from './SubwayLine';
+import { Subway } from './Subway';
+
+export function parse(data) {
+    let currentData = data;
+    return convertToSubway(currentData);
+} 
+
+function convertToSubway(data) {
+    let allNames = [];
+    let stationNames = [];
+    let subwayLines = [];
+    let result = {};
+    data = data.split('\n').map(name => name.trim());
+
+    for (let i = 0; i < data.length; i++) {
+        if (!data[i] || i === data.length - 1) {
+            if (i === data.length - 1) {
+                allNames.push(data[i]);
+            }
+            let lineName = allNames.shift();
+            stationNames = stationNames.concat(allNames);
```

Comment:
Why do you do concat? it seems that you do not need stationNames variable at all. Let's discuss the logic again